gangplank/ocp/filer.go

```diff
@@ -258,15 +259,53 @@ func (m *minioServer) exec(ctx context.Context) error {
 	}
 
 	// Ensure the process gets terminated on shutdown
+	sigs := make(chan os.Signal, 64)
 	go func() {
-		sigs := make(chan os.Signal, 1)
 		signal.Notify(sigs, os.Interrupt, syscall.SIGTERM, syscall.SIGUSR1, syscall.SIGUSR2)
 		<-sigs
 		m.Kill()
 	}()
 
+	readyFn := func() <-chan bool {
+		readyChan := make(chan bool)
+		go func() {
+			for {
+				c, _ := m.client()
+				client := http.Client{}
+
+				if m.Secure {
+					client.Transport = &http.Transport{
+						TLSClientConfig: tlsconfig.ClientDefault(),
+					}
+				}
+
+				urlS := c.EndpointURL().String()
+				resp, err := client.Head(urlS)
+
+				if err != nil || resp == nil {
+					continue
+				}
+				fmt.Println(resp.StatusCode)
+				// Ping the minio server to wait for it to come up. Accept any response.
+				if resp.StatusCode >= 200 && resp.StatusCode < 500 {
+					readyChan <- true
+					return
+				}
+			}
```

Comment:
Shouldn't we sleep somewhere in this loop between tries?

Reply:
I set the timeout to 5 seconds which should the same thing.